src/Service.Host/client/src/components/PcasStorageType.js

```diff
@@ -0,0 +1,280 @@
+                        propertyName="maximum"
+                        label="Maximum"
+                        type="number"
+                        value={pcasStorageType?.maximum}
+                        fullWidth
+                        onChange={handleFieldChange}
+                    />
+                </Grid>
+                <Grid size={3}>
+                    <InputField
+                        propertyName="increment"
+                        label="Incr"
+                        type="number"
+                        value={pcasStorageType?.increment}
+                        fullWidth
+                        onChange={handleFieldChange}
+                    />
+                </Grid>
+                <Grid size={3}>
+                    <InputField
+                        propertyName="preference"
+                        label="Preference"
+                        value={pcasStorageType?.preference}
+                        fullWidth
+                        onChange={handleFieldChange}
+                    />
+                </Grid>
+            </Grid>
+            <Grid size={12}>
+                <InputField
+                    propertyName="remarks"
+                    label="Remarks"
+                    value={pcasStorageType?.remarks}
+                    fullWidth
+                    onChange={handleFieldChange}
+                />
+            </Grid>
+            <Grid container spacing={3}>
+                <Grid size={12}>
+                    <Button
+                        variant="contained"
+                        fullWidth
+                        disabled={
+                            pcasStorageType === pcasStorageTypeResult ||
+                            !pcasStorageType?.boardCode ||
+                            !pcasStorageType?.storageTypeCode
+                        }
+                        onClick={() => {
+                            if (creating) {
+                                createPartStorageType(null, pcasStorageType);
+                            } else {
+                                updatePartStorageType(
+                                    `${boardCode}/${storageTypeCode}`,
+                                    pcasStorageType
+                                );
+                            }
+                        }}
+                    >
+                        {creating ? 'Create ' : 'Save'}
+                    </Button>
+                </Grid>
+                <Grid size={12}>
+                    <SnackbarMessage
+                        visible={snackbarVisible}
+                        onClose={() => setSnackbarVisible(false)}
+                        message="Save Successful"
+                    />
+                </Grid>
+                {(updateError || createError) && (
+                    <Grid size={12}>
+                        <ErrorCard
+                            errorMessage={updateError ? updateError?.details : createError}
```

Comment:
i assume createError and updateError have the same shape, so seems odd you'd need to access the .details field of one but not the other?